### PR TITLE
Book detail

### PR DESCRIPTION
**Pull request by Gianni-Nicolas**
**Summary**

The bookDetail.html file in which the details were written was added and the image of a book with the corresponding styles was placed, according to the card Detail of book 1, Detail of book 2 and Detail of book 3.

**Screenshot**

Captura de Pantalla 2020-02-07 a la(s) 11 54 00

**Trello Card**

- https://trello.com/c/GH7EgnOH/26-detalle-de-libro-i-textos

- https://trello.com/c/K4G6aUHp/27-detalle-de-libro-ii-imagen

- https://trello.com/c/KvB6oCNE/28-detalle-de-libro-iii-estructura

### CODE REVIEW

File: bookDetail.html [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    <div class="general-container">
+        <div class="container-book">
+            <img class="book-cover" src="images/book-cover.png" />
+            <div class="book-information">
+                <header class="header-book">
+                    <h1 class="title">Titulo del libro <span class="title-description">(género)</span></h1>
+                    <div class="underline"></div>
+                </header>
+                <section class="section-book">
+
```
Comment (idcuadrowolox) [inline bookDetail.html]: Remove empty line

File: bookDetail.html [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+                    <p class="detail">Autor del libro: <span class="description">Nombre del autor del libro</span></p>
+
```
Comment (idcuadrowolox) [inline bookDetail.html]: Remove empty line

File: bookDetail.html [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+                    <p class="detail">Editorial: <span class="description">Nombre de la editorial</span></p>
+
```
Comment (idcuadrowolox) [inline bookDetail.html]: Remove empty line

File: bookDetail.html [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                    <div class="underline"></div>
+                </header>
+                <section class="section-book">
+
+                    <p class="detail">Autor del libro: <span class="description">Nombre del autor del libro</span></p>
+
+                    <p class="detail">Editorial: <span class="description">Nombre de la editorial</span></p>
+                    
+                    <p class="detail">Año de publicación: <span class="description">Año publicación</span> </p>
+
```
Comment (idcuadrowolox) [inline bookDetail.html]: Remove empty line

File: bookDetail.html [hunk 5]
```diff
@@ -0,0 +1,33 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title>HTML & CSS Training</title>
+  <meta name="description" content="training">
+  <meta name="author" content="Wolox">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
```
Comment (idcuadrowolox) [inline bookDetail.html]: fix indentation

File: bookDetail.html [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+  <link rel="stylesheet"  type="text/css" href="css/application.css">
+</head>
+<body>
+    <div class="general-container">
+        <div class="container-book">
+            <img class="book-cover" src="images/book-cover.png" />
+            <div class="book-information">
+                <header class="header-book">
+                    <h1 class="title">Titulo del libro <span class="title-description">(género)</span></h1>
+                    <div class="underline"></div>
```
Comment (idcuadrowolox) [inline bookDetail.html]: Add the underline to h1 tag

**Comment by martinmona** [comment]
Remember to add a screenshot

File: bookDetail.html [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+                </header>
+                <section class="section-book">
+
```
Comment (martinmona) [inline bookDetail.html]: delete empty spaces

File: bookDetail.html [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+                    <p class="detail">Autor del libro: <span class="description">Nombre del autor del libro</span></p>
+
```
Comment (martinmona) [inline bookDetail.html]: same

File: bookDetail.html [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+<head>
+  <meta charset="UTF-8">
+  <title>HTML & CSS Training</title>
+  <meta name="description" content="training">
+  <meta name="author" content="Wolox">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet"  type="text/css" href="css/application.css">
+</head>
+<body>
+    <div class="general-container">
```
Comment (martinmona) [inline bookDetail.html:12]: Indentation is 2 spaces

File: bookDetail.html [hunk 10]
```diff
@@ -0,0 +1,34 @@
...
+        <div class="container-book">
+            <img class="book-cover" src="images/book-cover.png" />
+            <div class="book-information">
+                <header class="header-book">
+                    <h1 class="title">Titulo del libro <span class="title-description">(género)</span></h1>
+                    <div class="underline"></div>
```
Comment (martinmona) [inline bookDetail.html]: avoid using unnecessary tags, you can use `border-bottom`

File: scss/detail.scss [hunk 11]
```diff
@@ -0,0 +1,90 @@
+@import 'colors';
+
+%display-config-vertical {
+    display: flex;
```
Comment (martinmona) [inline scss/detail.scss]: remember 2 spaces indentation

File: scss/detail.scss [hunk 12]
```diff
@@ -0,0 +1,89 @@
...
+    color: $description-book-color;
+    font-size: 24px;
+    font-weight: 700;
+    line-height: 28px;
+  }
+
+.underline{
+    height: 4px;
+    width: 95%;
+    background-color: #BED23C;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: use scss variables for colors

File: scss/detail.scss [hunk 13]
```diff
@@ -0,0 +1,89 @@
+@import 'colors';
```
Comment (idcuadrowolox) [inline scss/detail.scss]: use 2 spaces for indent

File: scss/detail.scss [hunk 14]
```diff
@@ -0,0 +1,89 @@
...
+    display: flex;
+    align-items: center;
+}
+
+%conteiner-book-elements-height {
+    max-height: 90%;
+    height: 90%;
+}
+
+* {
```
Comment (idcuadrowolox) [inline scss/detail.scss:15]: I think you can put this in application.scss

File: scss/detail.scss [hunk 15]
```diff
@@ -0,0 +1,89 @@
...
+    box-shadow: 4px 4px 10px 0 rgba(0,0,0,0.5);
+    @extend %display-config-vertical;
+    justify-content: flex-start;
+}
+
+.book-cover {
+    @extend %conteiner-book-elements-height;
+    max-width: 25%;
+    width: 25%;
+    margin: 3%;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: put padding in parent container

File: scss/detail.scss [hunk 16]
```diff
@@ -0,0 +1,89 @@
...
+    height: 60%;
+    width: 75%;
+    background-color: white;
+    box-shadow: 4px 4px 10px 0 rgba(0,0,0,0.5);
+    @extend %display-config-vertical;
+    justify-content: flex-start;
+}
+
+.book-cover {
+    @extend %conteiner-book-elements-height;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: unnecessary property, when you work with images is a good practice to set only with o height.

File: bookDetail.html [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+                </header>
+                <section class="section-book">
```
Comment (idcuadrowolox) [inline bookDetail.html]: Remove container

File: scss/detail.scss [hunk 18]
```diff
@@ -0,0 +1,89 @@
...
+    height: 100vh;
+}
+
+.container-book {
+    height: 60%;
+    width: 75%;
+    background-color: white;
+    box-shadow: 4px 4px 10px 0 rgba(0,0,0,0.5);
+    @extend %display-config-vertical;
+    justify-content: flex-start;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: please, order alphabetically

File: scss/colors.scss [hunk 19]
```diff
@@ -0,0 +1,3 @@
+$general-background-color: #F4F4F4;
+$detail-book-color: #000;
```
Comment (idcuadrowolox) [inline scss/colors.scss]: I think it's better to use variables like $black, $main-green for example

File: scss/detail.scss [hunk 20]
```diff
@@ -0,0 +1,89 @@
...
+.general-container {
+    background-color: $general-background-color;
+    @extend %display-config-vertical;
+    justify-content: center;
+    height: 100vh;
+}
+
+.container-book {
+    height: 60%;
+    width: 75%;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: work responsive views with media queries

**Review by idcuadrowolox — CHANGES_REQUESTED** [review]

File: bookDetail.html [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+  <title>HTML & CSS Training</title>
+  <meta name="description" content="training">
+  <meta name="author" content="Wolox">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet"  type="text/css" href="css/application.css">
+</head>
+<body>
+    <div class="general-container">
+        <div class="container-book">
+            <img class="book-cover" src="images/book-cover.png" />
```
Comment (idcuadrowolox) [inline bookDetail.html]: It's a good practice to add the alt attribute to image tags

**Review by idcuadrowolox — CHANGES_REQUESTED** [review]

File: scss/detail.scss [hunk 22]
```diff
@@ -0,0 +1,90 @@
...
+.section-book {
+    display: flex;
+    flex-flow: column wrap;
+}
+
+.detail {
+    color: $detail-book-color;
+    font-size: 20px;
+    font-weight: 700;
+    line-height: 24px;
```
Comment (martinmona) [inline scss/detail.scss]: maybe you can use @mixin to change font-size, weight and line height

File: scss/colors.scss [hunk 23]
```diff
@@ -0,0 +1,5 @@
+$general-background-color: #F4F4F4;
```
Comment (martinmona) [inline scss/colors.scss]: > this color variables should describe how the color looks and not where it is used


Check the tech-guides, and [Name that color](http://chir.ag/projects/name-that-color/) page

**Review by martinmona — CHANGES_REQUESTED** [review]

File: scss/detail.scss [hunk 24]
```diff
@@ -46,26 +52,22 @@
...
   color: $grey;
-  font-size: 24px;
-	font-weight: 700;
-  height: 30px;
+  @include font-config(24px, 700, 30px);
   line-height: 28px;
   width: 98px;
 }
 
 .detail {
```
Comment (idcuadrowolox) [inline scss/detail.scss:77]: Please check indent in this class

File: scss/detail.scss [hunk 25]
```diff
@@ -11,6 +11,12 @@
   height: 368px;
 }
 
+@mixin font-config($size, $weight, $height) {
+  font-size: $size;
+	font-weight: $weight;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: please check indent

File: scss/detail.scss [hunk 26]
```diff
@@ -63,15 +63,15 @@
...
 .detail {
-	color: $detail-book-color;
+	color: $black;
 	font-size: 20px;
 	font-weight: 700;
 	line-height: 24px;
 	margin-bottom: 40px;
 }
 
 .description {
```
Comment (idcuadrowolox) [inline scss/detail.scss:84]: please check indent in this class

File: scss/detail.scss [hunk 27]
```diff
@@ -54,7 +54,7 @@
 }
 
 .title-description {
-  color: $description-book-color;
+  color: $grey;
   font-size: 24px;
 	font-weight: 700;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: check Indent

File: scss/detail.scss [hunk 28]
```diff
@@ -16,7 +16,7 @@
 }
 
 .general-container {
-  background-color: $general-background-color;
+  background-color: $light-grey;
   @extend %display-config-vertical;
 	height: 100vh;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: please check indent

File: scss/detail.scss [hunk 29]
```diff
@@ -16,7 +16,7 @@
...
 	justify-content: center;
```
Comment (idcuadrowolox) [inline scss/detail.scss]: please check indent

File: scss/colors.scss [hunk 30]
```diff
@@ -1,5 +1,5 @@
-$general-background-color: #F4F4F4;
-$detail-book-color: #000;
-$description-book-color: #828282;
-$underline-color: #BED23C;
+$light-grey: #F4F4F4;
+$black: #000;
+$grey: #828282;
+$green: #BED23C;
```
Comment (idcuadrowolox) [inline scss/colors.scss:3]: Order alphabetically

**Review by idcuadrowolox — CHANGES_REQUESTED** [review]

**Review by idcuadrowolox — APPROVED** [review]

File: scss/detail.scss [hunk 31]
```diff
@@ -0,0 +1,79 @@
...
+
+%display-config-vertical {
+  align-items: center;
+  display: flex;
+}
+
+%container-book-elements-height {
+  height: 368px;
+  max-height: 100%;
+  height: 368px;
```
Comment (nvescio) [inline scss/detail.scss]: Check here, you've `height` twice

**Review by nvescio — CHANGES_REQUESTED** [review]

**Review by nvescio — APPROVED** [review]

**Review by martinmona — APPROVED** [review]

File: scss/detail.scss [hunk 32]
```diff
@@ -0,0 +1,79 @@
...
+
+* {
+  font-family: "Roboto", sans-serif;
+}
+
+.general-container {
+  background-color: $light-grey;
+  @extend %display-config-vertical;
+  height: 100vh;
+  justify-content: center;
```
Comment (martinmona) [inline scss/detail.scss]: Not necessary to use flex here, if you want to center the `container-book` class you can use `margin: 0 auto` there

File: scss/detail.scss [hunk 33]
```diff
@@ -0,0 +1,79 @@
...
+
+.book-cover {
+  @extend %container-book-elements-height;
+  margin-right: 30px;
+}
+
+.book-information {
+  display: flex;
+  @extend %container-book-elements-height;
+  flex-flow: column nowrap;
```
Comment (martinmona) [inline scss/detail.scss]:
```suggestion
  flex-direction: column;
```

File: scss/detail.scss [hunk 34]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.title {
+  border-bottom: 4px solid $green;
+  color: $black;
+  @include font-config(32px, 700);
+  margin-bottom: 30px;
+  padding-bottom: 8px;
+  width: 500px;
```
Comment (martinmona) [inline scss/detail.scss:67]: add `max-width: 100%`

File: scss/detail.scss [hunk 35]
```diff
@@ -0,0 +1,79 @@
...
+}
+
+.title-description {
+  color: $grey;
+  @include font-config(24px, 700);
+  height: 30px;
+  line-height: 28px;
+  width: 98px;
```
Comment (martinmona) [inline scss/detail.scss]: maybe it's not necessary

File: bookDetail.html [hunk 36]
```diff
@@ -0,0 +1,24 @@
...
+  <title>HTML & CSS Training</title>
+  <meta name="description" content="training">
+  <meta name="author" content="Wolox">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet" type="text/css" href="css/application.css">
+</head>
+<body>
+  <div class="general-container">
+    <div class="container-book">
+      <img class="book-cover" src="images/book-cover.png" alt="book-cover"/>
```
Comment (martinmona) [inline bookDetail.html]: Create an `assets` folder and move this image there

File: scss/detail.scss [hunk 37]
```diff
@@ -0,0 +1,79 @@
...
+  @include font-config(32px, 700);
+  margin-bottom: 30px;
+  padding-bottom: 8px;
+  width: 500px;
+}
+
+.title-description {
+  color: $grey;
+  @include font-config(24px, 700);
+  height: 30px;
```
Comment (martinmona) [inline scss/detail.scss]: you are already using `line-height` instead

**Review by martinmona — APPROVED** [review]

**Review by martinmona — APPROVED** [review]

**Pull request merged**